packages/medusa/src/api/routes/store/products/get-product.ts

```diff
@@ -76,7 +79,7 @@ import { validator } from "../../../../utils/validator"
 export default async (req, res) => {
   const { id } = req.params
 
-  const validated = await validator(PriceSelectionParams, req.query)
+  const validated = await validator(StoreGetProductParams, req.query)
```

Comment:
Todo: if we apply the transformer, the handler should be cleaned up like the others. If you agree and you need, we can look together

Reply:
@adrien2p Did you have anything to add?

Reply:
Sorry, I mean that there should be any use of the validator in the handler as it should be managed by the transformer middleware